corehq/form_processor/management/commands/fix_edit_forms_attachments.py

```diff
@@ -0,0 +1,121 @@
+            print('scanning %s' % db_alias)
+            edit_form_ids = (XFormOperationSQL.objects.using(db_alias)
+                             .filter(operation='edit')
+                             .values_list('form_id', flat=True))
+            for edit_form_id in with_progress_bar(edit_form_ids):
+                edited_form = form_accessor.get_form(edit_form_id)
+                deprecated_form_id = edited_form.deprecated_form_id
+                original_attachments = list(FormAccessorSQL.get_attachments_for_forms([deprecated_form_id]))
+                new_attachments = list(FormAccessorSQL.get_attachments_for_forms([edit_form_id]))
+                new_attachment_names = [a.name for a in new_attachments]
+                for attachment in original_attachments:
+                    if attachment.name not in new_attachment_names:
+                        assert attachment.form_id == deprecated_form_id
+                        add_forms_with_attachments[edit_form_id].append(
+                            (attachment.attachment_id, deprecated_form_id)
+                        )
+        return add_forms_with_attachments
+
+    @staticmethod
+    def _find_couch_forms_with_missing_attachments():
+        """
+        find deprecated form ids,
+        then look for the new form and compare attachments between them
+        :return:
+        """
+        add_forms_with_attachments = defaultdict(list)
+        deprecated_form_ids = [
+            f['id'] for f in XFormDeprecated.view(
+                'all_docs/by_doc_type', startkey=['XFormDeprecated'], endkey=['XFormDeprecated', {}],
+                include_docs=False, reduce=False).all()]
+        for deprecated_form_id in deprecated_form_ids:
+            deprecated_form = XFormDeprecated.get(deprecated_form_id)
+            form_accessor = FormAccessors(deprecated_form.domain)
+            if deprecated_form.orig_id:
+                try:
+                    new_form = form_accessor.get_form(deprecated_form.orig_id)
+                except XFormNotFound:
+                    # fall back to other DB couch/sql to look for the form
+                    if form_accessor.db_accessor == FormAccessorSQL:
+                        new_form = FormAccessorCouch.get_form(deprecated_form.orig_id)
+                    else:
+                        new_form = FormAccessorSQL.get_form(deprecated_form_id)
+                except ValueError as e:
+                    # always is illegal doc id error
+                    # https://github.com/dimagi/commcare-hq/blob/652089eb7b63e3967d674580cab55522b5327a22/corehq/blobs/mixin.py#L614
+                    print(e)
+                    continue
+                original_attachments = deprecated_form.attachments
+                new_attachments = new_form.attachments
+                new_attachment_names = new_attachments.keys()
+                for attachment_name in original_attachments:
+                    if attachment_name not in new_attachment_names:
```

Comment:
> Why does this seem like a non-robust check - are there other reasons that a form would not have the same attachments as its parent?

I don't know for sure, but I wouldn't be surprised given that form deprecation seems like an edge case that probably isn't always handled consistently in the code. I would prefer not to update thousands of old deprecated forms just because they meet this criteria.

Maybe as a compromise, only update forms that have been created since the new "clean forms" feature was implemented?

Reply:
Did a small test for couch deprecated forms i found
```
In [62]: len(form_ids)
Out[62]: 2717

In [63]: form_ids[0]
Out[63]: u'ad1257dacfd4dad7343998e3994a087e'

In [64]: release_date
Out[64]: datetime.date(2018, 5, 14)

In [65]: today = datetime.datetime.today().date()

In [66]: today
Out[66]: datetime.date(2018, 11, 7)

In [67]: after_release_not_today = []

In [68]: for form_id in form_ids:
    ...:     try:
    ...:         form = FormAccessorCouch.get_form(form_id)
    ...:     except XFormNotFound:
    ...:         form = FormAccessorSQL.get_form(form_id)
    ...:     form_deprecated_date = form.deprecated_date.date()
    ...:     if (form_deprecated_date >= release_date and
    ...:             form_deprecated_date != datetime.datetime.today().date()):
    ...:         after_release_not_today.append(form_id)
    ...:

In [69]: len(after_release_not_today)
Out[69]: 0
```
ignored today because it defaults to now. 
For example [this](https://www.commcarehq.org/hq/admin/raw_doc/?id=c37b1e69657833974a6651316b6cd89c&db_name=) deprecated form

Does this mean that no couch form that was edited has this issue?

Reply:
Maybe? I'm not sure how to evaluate if that seems reasonable or not. Maybe @orangejenny would know?

Reply:
similar point here
https://github.com/dimagi/commcare-hq/pull/22313#discussion_r232505695

Reply:
> Does this mean that no couch form after the feature release, has this issue?

That's plausible. I'd expect the set of domains that collect multimedia in forms is fairly small, and I'd also expect that a minority of domains have used data cleaning.